quickwit/quickwit-search/src/cluster_client.rs

```diff
@@ -157,6 +170,57 @@ impl ClusterClient {
         // TODO: implement retry
         client.leaf_list_terms(request.clone()).await
     }
+
+    /// Attempts to store a given search context within the cluster.
+    ///
+    /// This function may fail silently, if no clients was available.
+    pub async fn put_kv(&self, key: &[u8], payload: &[u8], ttl: Duration) {
+        let clients = self
+            .search_job_placer
+            .best_nodes_per_affinity(key)
+            .await
+            .take(MAX_PUT_KV_ATTEMPTS);
+
+        // We run the put requests concurrently.
+        // Our target is a replication over TARGET_NUM_REPLICATION nodes, we therefore try to avoid
+        // replicating on more than TARGET_NUM_REPLICATION nodes at the same time. Of
+        // course, this may still result in the replication over more nodes, but this is not
+        // a problem.
+        //
+        // The requests are made in a concurrent manner, up to two at a time. As soon as 2 requests
+        // are successful, we stop.
+        let successful_replication = futures::stream::iter(clients)
+            .map(|mut client| {
+                let put_kv_request = PutKvRequest {
+                    key: key.to_vec(),
+                    payload: payload.to_vec(),
+                    ttl_secs: ttl.as_secs() as u32,
+                };
+                async move { client.put_kv(put_kv_request).await }
+            })
+            .buffer_unordered(TARGET_NUM_REPLICATION)
+            .filter(|put_kv_successful| ready(put_kv_successful.is_ok()))
+            .take(TARGET_NUM_REPLICATION)
+            .count()
+            .await;
+
+        if successful_replication == 0 {
```

Comment:
nit: I'd display a warning if any attempt failed and an error if all of them failed. It would be also nice to display the number of failures and which nodes failed. Given the urgency of this PR, feel free to ignore this comment.

Reply:
excellent point.

I went for warn whenever an attempt failed, error if all failed and there was at least one node.